packages/backend-auth/src/userpool_access_policy_factory.ts

```diff
@@ -0,0 +1,122 @@
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Stack } from 'aws-cdk-lib';
+import { AmplifyFault, AmplifyUserError } from '@aws-amplify/platform-core';
+import { ActionIam, ActionMeta, AuthActions } from './types.js';
+
+/**
+ * Generates IAM policies scoped to a single userpool.
+ */
+export class UserPoolAccessPolicyFactory {
+  private readonly namePrefix = 'userpoolAccess';
+  private readonly stack: Stack;
+
+  private policyCount = 1;
+
+  /**
+   * Instantiate with the userpool to generate policies for
+   */
+  constructor(private readonly userpool: IUserPool) {
+    this.stack = Stack.of(userpool);
+  }
+
+  createPolicy = (actions: AuthActions) => {
+    const policyActions: Set<string> = new Set();
+
+    if (actions.length === 0) {
+      throw new AmplifyUserError('EmptyPolicyError', {
+        message: 'At least one action must be specified',
+      });
+    }
+
+    actions.forEach((authAction) => {
+      const mappedAction = iamActionMap[authAction];
+
+      if (typeof mappedAction === 'string') {
+        policyActions.add(mappedAction);
+      } else {
+        mappedAction.forEach((action) => policyActions.add(action));
+      }
+    });
+
+    if (policyActions.size === 0) {
+      throw new AmplifyFault('EmptyPolicyFault', {
+        message: 'Failed to construct valid policy to access UserPool',
+      });
+    }
+
+    const policy = new Policy(
+      this.stack,
+      `${this.namePrefix}${this.policyCount++}`,
+      {
+        statements: [
+          new PolicyStatement({
+            actions: [...policyActions],
+            resources: [this.userpool.userPoolArn],
+          }),
+        ],
+      }
+    );
+
+    return policy;
+  };
+}
+
+type IamActionMap = {
+  [action in ActionIam]: string;
+} & {
+  [amplifyActionSet in ActionMeta]: string[];
+};
+
+const iamActionMap: IamActionMap = {
+  manageUsers: [
+    'cognito-idp:AdminConfirmSignUp',
+    'cognito-idp:AdminCreateUser',
+    'cognito-idp:AdminDeleteUser',
+    'cognito-idp:AdminDeleteUserAttributes',
+    'cognito-idp:AdminDisableUser',
+    'cognito-idp:AdminEnableUser',
+    'cognito-idp:AdminGetUser',
+    'cognito-idp:AdminListGroupsForUser',
+    'cognito-idp:AdminRespondToAuthChallenge',
+    'cognito-idp:AdminSetUserMFAPreference',
+    'cognito-idp:AdminSetUserSettings',
+    'cognito-idp:AdminUpdateUserAttributes',
+    'cognito-idp:AdminUserGlobalSignOut',
```

Comment:
Same here, suggest this as a standalone action only

Reply:
@edwardfoyle are you thinking a standalone meta action? We've bucketed every individual action so far

Reply:
No, just take it out of "manageUsers" but leave it [here](https://github.com/aws-amplify/amplify-backend/pull/1090/files#diff-0141bb8674518ff023c7a33a2bf8bef5a8189f0c6c1ba39558a10195dbc5b886R121)

Reply:
perhaps we should remove this one entirely?

Reply:
Sure